server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CellIdSource.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.search.aggregations.bucket.composite;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.elasticsearch.index.fielddata.AbstractSortingNumericDocValues;
+import org.elasticsearch.index.fielddata.MultiGeoPointValues;
+import org.elasticsearch.index.fielddata.SortedBinaryDocValues;
+import org.elasticsearch.index.fielddata.SortedNumericDoubleValues;
+import org.elasticsearch.search.aggregations.support.ValuesSource;
+
+import java.io.IOException;
+
+/**
+ * Wrapper class to help convert {@link MultiGeoPointValues}
+ * to numeric long values for bucketing.
+ */
+class CellIdSource extends ValuesSource.Numeric {
```

Comment:
This class was copied over from `org.elasticsearch.search.aggregations.bucket.geogrid.CellIdSource`. I could just make that other class `public` and access it from the `composite` package. I am not sure what is the best practice within the aggs framework. 

the difference between this class and the one stored in `geogrid` is this one allows access to the `encoder` which is needed when constructing the `GeohashValuesSource` object.